scio-jdbc/src/main/scala/com/spotify/scio/jdbc/sharded/JdbcShardable.scala

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2020 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio.jdbc.sharded
+
+import java.sql.ResultSet
+import org.slf4j.LoggerFactory
+
+trait JdbcShardable[T] extends Serializable {
+
+  def columnValueDecoder(resultSet: ResultSet, columnName: String): T
+
+  def partition(range: Range[T], numShards: Int): Seq[ShardQuery]
+
+}
+
+object ShardBy {
+
+  object range {
```

Comment:
what about moving this under the companion object? `Shard.range[T]` / `Shard.prefix[T](p)`